xbmc/utils/StringUtils.cpp

```diff
@@ -1889,6 +1889,20 @@ std::string StringUtils::FormatFileSize(uint64_t bytes)
   return Format("{:.{}f}{}", value, decimals, units[i]);
 }
 
+bool StringUtils::Contains(std::string_view str, std::string_view keyword, bool isCaseInsensitive)
+{
+  if (isCaseInsensitive)
+  {
+    auto itStr = std::search(str.begin(), str.end(), keyword.begin(), keyword.end(),
+                             [](unsigned char ch1, unsigned char ch2) {
+                               return std::toupper(ch1) == std::toupper(ch2);
+                             });
+    return (itStr != str.end());
+  }
+
+  return std::strstr(str.data(), keyword.data()) != nullptr;
```

Comment:
`std::strstr` expects the strings to be null terminated which `std::string_view` isn't. You can use `std::string_view::find` instead.

Reply:
thanks